cabal-install/cabal-install.cabal

```diff
@@ -307,7 +307,7 @@ executable cabal
         containers >= 0.5      && < 0.7,
         cryptohash-sha256 >= 0.11 && < 0.12,
         deepseq    >= 1.3      && < 1.5,
-        directory  >= 1.2.2.0  && < 1.4,
+        directory  >= 1.3.3.0  && < 1.4,
```

Comment:
That's a fairly recent `directory` version; can we make this conditional on `if os(windows)`?

Reply:
I don't understand the problem. Are you worried about the stability of recent versions of `directory`?

This change uses the `createFileLink` function, which was added in `directory-1.3.1.0`, so the lower bound will need to be raised anyways. Is `1.3.1.0` an acceptable lower bound? One potential downside is that the behaviour of `new-install` on Windows will differ depending on which version of `directory` was used to compile `cabal-install`.

Reply:
Well, we have all those `*.Compat.*` modules for a reason... ;-)

See 

- https://github.com/haskell/cabal/tree/master/Cabal/Distribution/Compat
- https://github.com/haskell/cabal/tree/master/cabal-install/Distribution/Client/Compat

So I'd strongly suggest to try to follow the prior-art from those compat modules to avoid requiring such a bleeding edge `directory` version